### PR TITLE
apollo-datasource-rest RESTDataSource does not handle HTTP 204 responses correctly

### PR DESCRIPTION
**Pull request by dhm116**
We noticed that the [RESTDataSource.parseBody](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-datasource-rest/src/RESTDataSource.ts#L104..L115) method was not handling HTTP 204 responses correctly in the event that a `Content-Type: application/json` header was being defined on the response.

This PR resolves that issue by inspecting the `response.status` to ensure that it is not `204` before attempting to process the response as JSON.

TODO:

* [x] Update CHANGELOG.md with your change (include reference to issue & this PR)
* [x] Make sure all of the significant new logic is covered by tests
* [x] Rebase your changes on master so that they can be merged easily
* [x] Make sure all tests and linter rules pass

### CODE REVIEW

**Comment by jplock** [comment]
We ran into this issue with an API behind AWS API Gateway where we were returning a 204 status code with no Content-Type response header.  API Gateway seems to be adding a `Content-Type: application/json` response header if one isn't defined and we haven't yet figured out a way to disable that.  To @dhm116's point, the `Content-Length` header value is properly being set to zero.

**Comment by dhm116** [comment]
I'll get the tests fixed up today!

File: packages/apollo-datasource-rest/src/__tests__/RESTDataSource.test.ts [hunk 1]
```diff
@@ -31,13 +31,13 @@ describe('RESTDataSource', () => {
 
   describe('constructing requests', () => {
     it('interprets paths relative to the base URL', async () => {
-      const dataSource = new (class extends RESTDataSource {
+      const dataSource = new class extends RESTDataSource {
```
Comment (dhm116) [inline packages/apollo-datasource-rest/src/__tests__/RESTDataSource.test.ts]: I'm not really sure how/why my PR failed to pass the linter check for this file after updating from `master` if the other updates to `master` were also merged in but somehow didn't fail this check?

As you can see, [I didn't modify these lines at all](https://github.com/apollographql/apollo-server/blob/867a94f2a6eb65243423b6bfe43aff61f1f25df7/packages/apollo-datasource-rest/src/__tests__/RESTDataSource.test.ts#L34), yet somehow my linter checks were failing after #2572 was merged in despite that passing.

**Comment by bartverbruggen** [comment]
This is fixed in https://github.com/apollographql/apollo-server/pull/2446

**Pull request closed without merging**